### PR TITLE
Add support for retrieving all addresses

### PR DESCRIPTION
**Pull request by nwithan8**
Finally, the C# library supports the ability to get all addresses with `Address.All()`

This PR includes:
- The new `Address.All()` function to retrieve all addresses. Can optionally pass in a `Dictionary<string, object>` as parameter. Function returns an `AddressCollection` object.
- Docstring for this function.
- Added new `AddressCollection` class, which features:
  - `public List<Batch> addresses`
  - `public bool has_more`
  This is similar to how our other client libraries handle multiple addresses.
- A new `AddressTest.TestRetrieveAll()` test for this feature

Classes, functions and tests have been imported/linked to all projects/versions, and all tests pass.

### CODE REVIEW

**Review by jchen293 — APPROVED** [review]

**Comment by jchen293** [comment]
The code looks good. Can you update the changelog as well?

File: EasyPost.Tests/AddressTest.cs [hunk 1]
```diff
@@ -246,6 +246,18 @@ public void TestInstanceCreateWithVerifications()
         [ExpectedException(typeof(HttpException))]
         public void TestRetrieveInvalidId() => Address.Retrieve("not-an-id");
 
+        [TestMethod]
+        public void TestRetrieveAll()
+        {
+            AddressCollection addressCollection = Address.All();
+            Assert.IsNotNull(addressCollection);
+            if (addressCollection.addresses.Count > 0)
```
Comment (Justintime50) [inline EasyPost.Tests/AddressTest.cs:254]: This is fine, could also iterate the returned list and assert against each

**Review by Justintime50 — APPROVED** [review]

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Actually I'm going to take another look first thing tomorrow. I think there's more needed here.

File: EasyPost.Tests/AddressTest.cs (hunk 1 above)
Comment (nwithan8) [reply EasyPost.Tests/AddressTest.cs:254]: Is this blocking the merge?

Comment (Justintime50) [reply EasyPost.Tests/AddressTest.cs:254]: Nope, if I stamp it means we can proceed but any feedback regardless of who left it is highly encouraged to include 😉 I won't stop progress for little nit picks though. Some of these are learnings we can share as a team for the future.

**Review by Justintime50 — APPROVED** [review]

File: EasyPost.Tests/AddressTest.cs (hunk 1 above)
Comment (nwithan8) [reply EasyPost.Tests/AddressTest.cs]: Resolved in 7e048a5

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]

**Pull request merged**